### DISCUSSION TITLE
Reading multiple files from file input

### DISCUSSION THREAD

**Discussion by bennythemink** [pocketbase/pocketbase#5418] [Q&A] [answered]
Hello,

I have a custom script in the pb_hooks directory. In this script I have a URL registered and it is the destination for a form that allows the user to enter their name and optionally send one or many files using a single file input that has the "multiple" attribute set. The form data is sent as FormData, not JSON. I have confirmed that the request contains all files as expected.

I can read the first file using `const files = c.formFile('documents')`, but I cannot figure out how to read the other files? Are multiple files on the same file input not supported? No problem if not, just wondering if I'm not seeing anything obvious.

Many thanks for any help.

**Comment by ganigeorgiev** [comment, marked as answer]
`c.formFile` returns only the first multipart file header associated with the key (_it is an echo context helper that calls the std [Request.FormFile](https://pkg.go.dev/net/http#Request.FormFile)_).

If you want to read all files you can try parsing the form and reading the file headers from its `file` map, something like:

```js
const form = c.multipartForm()

// should be an array of the parsed multipart file headers
const documentsFileHeaders = form.file["documents"] || [];

// convert the multipart file headers into filesystem.File if you want to use the files as part of a RecordUpsert form
const documents = [];
for (const mfh of documentsFileHeaders) {
    documents.push($filesystem.fileFromMultipart(mfh))
}
// ... do something with documents
```

**Reply by bennythemink** [reply]
Great, thank you for that @ganigeorgiev!
